R/expect-condition.R

```diff
@@ -90,6 +94,34 @@ expect_condition <- function(object,
   invisible(act$val %||% act$cap)
 }
 
+capture_first_condition <- function(expr) {
+  first <- NULL
+  n <- 0L
+
+  tryCatch(
+    withCallingHandlers(expr, condition = function(cnd) {
+      n <<- n + 1L
+      if (n == 1L) {
```

Comment:
Instead of a counter this could be `if (is.null(first))` right?

Reply:
I tried that first, but it's uglier because then you have to figure out whether or not to resignal the error in the `tryCatch()`. And you can't just check the identity of the condition, because we entrace it in the calling handler.

Reply:
Ah good point. Another possibility is to use `return_from()` to shortcircuit control flow, instead of `tryCatch()`.

Reply:
Yeah, that works out much more cleanly.